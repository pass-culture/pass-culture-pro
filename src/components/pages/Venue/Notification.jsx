import React from 'react'
import { Link } from 'react-router-dom'

import { closeNotification } from 'store/reducers/notificationReducer'

const handleOnClick = dispatch => () => dispatch(closeNotification())
const NotificationMessage = ({ venueId, offererId, dispatch }) => {
  const createOfferPathname = `/offres/creation?lieu=${venueId}&structure=${offererId}`
  return (
    <p>
      {'Lieu créé. Vous pouvez maintenant y '}
      <Link
        onClick={handleOnClick(dispatch)}
        to={createOfferPathname}
      >
        {'créer une offre'}
      </Link>
      {', ou en importer automatiquement. '}
    </p>
  )
}

export default NotificationMessage
