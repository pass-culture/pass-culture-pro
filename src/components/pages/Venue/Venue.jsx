import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { Form } from 'react-final-form'
import { getCanSubmit, parseSubmitErrors } from 'react-final-form-utils'
import { NavLink } from 'react-router-dom'

import BankFieldsContainer from './BankFields/BankFieldsContainer'
import CreateOfferControl from './CreateOfferControl'
import { FRANCE_POSITION } from './GeoFields/positions'
import GeoFields from './GeoFields/GeoFields'
import IdentifierFields from './IdentifierFields/IdentifierFields'
import siretDecorator from './IdentifierFields/siretDecorator'
import ModifyOrCancelControl from './ModifyOrCancelControl/ModifyOrCancelControl'
import ReturnOrSubmitControl from './ReturnOrSubmitControl/ReturnOrSubmitControl'
import VenueProvidersManagerContainer from './VenueProvidersManager/VenueProvidersManagerContainer'
import HeroSection from 'components/layout/HeroSection'
import Icon from 'components/layout/Icon'
import Main from 'components/layout/Main'

const noop = () => {}

class Venue extends Component {
  constructor() {
    super()
    this.state = { isRequestPending: false }
  }

  componentDidMount() {
    const { handleInitialRequest } = this.props
    handleInitialRequest()
  }

  buildBackToInfos = (offererName, initialName, offererId) => {
    return {
      label: offererName === initialName ? 'STRUCTURE' : offererName,
      path: `/structures/${offererId}`,
    }
  }

  checkIfVenueExists = initialVenueId => {
    return !!initialVenueId
  }

  handleFormFail = formResolver => (state, action) => {
    const { handleSubmitRequestFail } = this.props
    const { payload } = action
    const nextState = { isRequestPending: false }
    const errors = parseSubmitErrors(payload.errors)
    handleSubmitRequestFail(state, action)
    this.setState(nextState, () => formResolver(errors))
  }

  handleFormSuccess = formResolver => (state, action) => {
    const {
      formInitialValues,
      handleSubmitRequestSuccess,
      history,
      match: {
        params: { offererId },
      },
      query,
    } = this.props
    const { id: venueId } = formInitialValues
    const nextState = { isRequestPending: false }
    const { isCreatedEntity } = query.context({ id: venueId })

    this.setState(nextState, () => {
      handleSubmitRequestSuccess(state, action)
      if (isCreatedEntity) {
        history.push(`/structures/${offererId}`)
      } else {
        query.changeToReadOnly(null, { id: venueId })
      }

      formResolver()
    })
  }

  onFormSubmit = formValues => {
    const { handleSubmitRequest } = this.props

    this.setState({ isRequestPending: true })

    const formSubmitPromise = new Promise(resolve => {
      handleSubmitRequest({
        formValues,
        handleFail: this.handleFormFail(resolve),
        handleSuccess: this.handleFormSuccess(resolve),
      })
    })

    return formSubmitPromise
  }

  render() {
    const {
      formInitialValues,
      match: {
        params: { offererId, venueId },
      },
      query,
      offerer,
    } = this.props
    const { name: offererName } = offerer || {}
    const {
      iban: initialIban,
      id: initialId,
      isVirtual: initialIsVirtual,
      name: initialName,
      siret: initialSiret,
      thumbCount: initialThumbCount,
    } = formInitialValues || {}
    const { isCreatedEntity, isModifiedEntity, readOnly } = query.context({
      id: venueId,
    })
    const { isRequestPending } = this.state

    return (
      <Main
        backTo={this.buildBackToInfos(offererName, initialName, offererId)}
        name="venue"
        handleDataRequest={noop}>
        <HeroSection subtitle={initialName} title="Lieu">
          {isCreatedEntity && (
            <p className="subtitle">Ajoutez un lieu où accéder à vos offres.</p>
          )}

          {this.checkIfVenueExists(initialId) && !isCreatedEntity && (
            <NavLink
              to={`/offres/creation?lieu=${initialId}`}
              className="cta button is-primary">
              <span className="icon">
                <Icon svg="ico-offres-w" />
              </span>
              <span>Créer une offre</span>
            </NavLink>
          )}
        </HeroSection>

        {!isCreatedEntity && (
          <VenueProvidersManagerContainer venue={formInitialValues} />
        )}

        {!initialIsVirtual && (
          <Form
            decorators={[siretDecorator]}
            initialValues={formInitialValues}
            name="venue"
            onSubmit={this.onFormSubmit}
            render={formProps => {
              const canSubmit = getCanSubmit(formProps)
              const { form, handleSubmit, values } = formProps
              const {
                latitude: formLatitude,
                longitude: formLongitude,
                selectedAddress: formSelectedAddress,
                siret: formSiret,
              } = values

              const siretValidOnCreation = formSiret && formSiret.length === 14
              const fieldReadOnlyBecauseFrozenFormSiretOnCreation =
                isCreatedEntity && siretValidOnCreation

              const siretValidOnModification =
                typeof initialSiret !== 'undefined'
              const fieldReadOnlyBecauseFrozenFormSiretOnModification =
                isModifiedEntity && siretValidOnModification

              const fieldReadOnlyBecauseFrozenFormSiret =
                fieldReadOnlyBecauseFrozenFormSiretOnCreation ||
                fieldReadOnlyBecauseFrozenFormSiretOnModification

              return (
                <form onSubmit={handleSubmit}>
                  <IdentifierFields
                    fieldReadOnlyBecauseFrozenFormSiret={
                      fieldReadOnlyBecauseFrozenFormSiret
                    }
                    formSiret={formSiret}
                    initialSiret={initialSiret}
                    isCreatedEntity={isCreatedEntity}
                    isModifiedEntity={isModifiedEntity}
                    readOnly={readOnly}
                    withComment={
                      (isCreatedEntity && !siretValidOnCreation) ||
                      (isModifiedEntity && !siretValidOnModification)
                    }
                    withSiret={isCreatedEntity || initialSiret}
                  />
                  <BankFieldsContainer
                    initialIban={initialIban}
                    initialThumbCount={initialThumbCount}
                    readOnly={readOnly}
                  />
                  <GeoFields
                    fieldReadOnlyBecauseFrozenFormSiret={
                      fieldReadOnlyBecauseFrozenFormSiret
                    }
                    initialSiret={initialSiret}
                    form={form}
                    formLatitude={
                      formLatitude === ''
                        ? FRANCE_POSITION.latitude
                        : formLatitude
                    }
                    formLongitude={
                      formLongitude === ''
                        ? FRANCE_POSITION.longitude
                        : formLongitude
                    }
                    formSelectedAddress={formSelectedAddress}
                    formSiret={formSiret}
                    isModifiedEntity={isModifiedEntity}
                    readOnly={readOnly}
                  />
                  <hr />
                  <div
                    className="field is-grouped is-grouped-centered"
                    style={{ justifyContent: 'space-between' }}>
                    <ModifyOrCancelControl
                      isCreatedEntity={isCreatedEntity}
                      offererId={offererId}
                      venueId={venueId}
                      readOnly={readOnly}
                    />
                    {readOnly && <CreateOfferControl venueId={venueId} />}
                    <ReturnOrSubmitControl
                      canSubmit={canSubmit}
                      isCreatedEntity={isCreatedEntity}
                      isRequestPending={isRequestPending}
                      offererId={offererId}
                      readOnly={readOnly}
                    />
                  </div>
                </form>
              )
            }}
          />
        )}
      </Main>
    )
  }
}

Venue.propTypes = {
  handleInitialRequest: PropTypes.func.isRequired,
  handleSubmitRequest: PropTypes.func.isRequired,
  handleSubmitRequestFail: PropTypes.func.isRequired,
  handleSubmitRequestSuccess: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired,
  query: PropTypes.object.isRequired,
}

export default Venue
