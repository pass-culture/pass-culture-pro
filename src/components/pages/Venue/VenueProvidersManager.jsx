import classnames from 'classnames'
import get from 'lodash.get'
import { Field, Form, Icon, mergeForm, SubmitButton } from 'pass-culture-shared'
import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import { NavLink } from 'react-router-dom'
import { compose } from 'redux'
import { requestData } from 'redux-saga-data'
import { selectCurrentUser } from 'with-login'

import VenueProviderItem from './VenueProviderItem'
import selectProviderById from 'selectors/selectProviderById'
import selectProviders from 'selectors/selectProviders'
import selectVenueProviderByVenueIdAndVenueProviderId from 'selectors/selectVenueProviderByVenueIdAndVenueProviderId'
import selectVenueProvidersByVenueId from 'selectors/selectVenueProvidersByVenueId'

class ProviderManager extends Component {
  constructor() {
    super()
    this.state = {
      withError: false,
    }
  }

  static getDerivedStateFromProps(nextProps) {
    const {
      match: {
        params: { venueProviderId },
      },
    } = nextProps
    const isNew = venueProviderId === 'nouveau'
    return {
      isNew,
    }
  }

  onAddClick = () => {
    const {
      history,
      match: {
        params: { offererId, venueId },
      },
    } = this.props
    history.push(
      `/structures/${offererId}/lieux/${venueId}/fournisseurs/nouveau`
    )
  }

  handleMergeForm = () => {
    const {
      dispatch,
      match: {
        params: { venueId },
      },
    } = this.props
    const { isNew } = this.state
    isNew && dispatch(mergeForm('venueProviders', 'nouveau', { venueId }))
  }

  handleDataRequest = () => {
    const {
      dispatch,
      match: {
        params: { venueId },
      },
    } = this.props
    dispatch(requestData({ apiPath: '/providers' }))
    dispatch(
      requestData({
        apiPath: `/venueProviders?venueId=${venueId}`,
        stateKey: 'venueProviders',
      })
    )
  }

  handleSuccess = () => {
    const {
      history,
      match: {
        params: { offererId, venueId },
      },
    } = this.props
    history.push(`/structures/${offererId}/lieux/${venueId}`)
  }

  componentDidMount() {
    const {
      match: {
        params: { venueProviderId },
      },
    } = this.props
    this.handleDataRequest()
    venueProviderId === 'nouveau' && this.handleMergeForm()
  }

  componentDidUpdate(prevProps) {
    if (
      prevProps.match.params.venueProviderId === 'nouveau' ||
      this.props.match.params.venueProviderId !== 'nouveau'
    ) {
      this.handleMergeForm()
    }
  }

  render() {
    const {
      match: {
        params: { offererId, venueId },
      },
      provider,
      providers,
      venue,
      venueProvider,
      venueProviders,
    } = this.props
    const { identifierDescription, identifierRegexp } = provider || {}
    const { isNew, withError } = this.state

    return (
      <div className="venue-providers-manager section">
        <h2 className="main-list-title">
          IMPORTATIONS D'OFFRES
          <span className="is-pulled-right is-size-7 has-text-grey">
            Si vous avez plusieurs comptes auprès de la même source, ajoutez-les
            successivement.
          </span>
        </h2>
        <ul
          className={classnames('main-list', {
            'is-marginless': !get(venueProviders, 'lenght') && !isNew,
          })}>
          {venueProviders.map((vp, index) => (
            <VenueProviderItem venue={venue} venueProvider={vp} key={vp.id} />
          ))}
          {isNew && (
            <li>
              <Form
                action="/venueProviders"
                className="level"
                handleSuccess={this.handleSuccess}
                name="venueProvider"
                patch={venueProvider}>
                <Field type="hidden" name="venueId" />
                {withError && (
                  <p
                    className={
                      withError ? 'has-text-weight-bold has-text-danger' : ''
                    }>
                    Il faut un identifiant ou celui-ci existe déjà
                  </p>
                )}
                <div className="level-left">
                  <div className="field picto level-item">
                    <Icon svg="picto-db-default" />
                  </div>
                  <Field
                    name="providerId"
                    options={providers}
                    optionValue="id"
                    placeholder="Source d\'importation"
                    required
                    size="small"
                    type="select"
                  />
                  {provider && identifierRegexp && (
                    <Field
                      name="venueIdAtOfferProvider"
                      placeholder="identifiant"
                      required
                      size="small"
                      title={identifierDescription}
                    />
                  )}
                </div>
                <div className="field level-item level-right">
                  <NavLink
                    className="button is-secondary"
                    to={`/structures/${offererId}/lieux/${venueId}`}>
                    Annuler
                  </NavLink>
                </div>
                {provider && (
                  <div className="field level-item level-right">
                    <SubmitButton className="button is-secondary">
                      Importer
                    </SubmitButton>
                  </div>
                )}
                <div />
              </Form>
            </li>
          )}
        </ul>
        <div className="has-text-centered">
          <button
            className="button is-secondary"
            disabled={isNew}
            onClick={this.onAddClick}>
            + Importer des offres
          </button>
        </div>
      </div>
    )
  }
}

function mapStateToProps(state, ownProps) {
  const { venue } = ownProps
  const { id: venueId } = venue || {}
  const providers = selectProviders(state)

  const formPatch = get(state, 'form.venueProvider')

  let provider
  if (providers.length === 1) {
    provider = providers[0]
  } else {
    const providerId = get(formPatch, 'providerId')
    provider = selectProviderById(state, providerId)
  }

  return {
    currentUser: selectCurrentUser(state),
    provider,
    providers,
    venueProvider: selectVenueProviderByVenueIdAndVenueProviderId(
      state,
      venueId
    ),
    venueProviders: selectVenueProvidersByVenueId(state, venueId),
  }
}

export default compose(
  withRouter,
  connect(mapStateToProps)
)(ProviderManager)
