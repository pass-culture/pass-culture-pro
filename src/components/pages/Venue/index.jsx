import get from 'lodash.get'
import { withLogin } from 'pass-culture-shared'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import { compose } from 'redux'

import offererSelector from '../../../selectors/offerer'
import selectUserOffererByOffererIdAndUserIdAndRightsType from '../../../selectors/selectUserOffererByOffererIdAndUserIdAndRightsType'
import selectVenuePatchByVenueIdByOffererId from '../../../selectors/selectVenuePatchByVenueIdByOffererId'

import RawVenuePage from './RawVenuePage'

function mapStateToProps(state, ownProps) {
  const { offererId, venueId } = ownProps.match.params
  const { id: userId } = state.user || {}
  return {
    adminUserOfferer: selectUserOffererByOffererIdAndUserIdAndRightsType(
      state,
      offererId,
      userId,
      'admin'
    ),
    formGeo: get(state, 'form.venue.geo'),
    formLatitude: get(state, 'form.venue.latitude'),
    formLongitude: get(state, 'form.venue.longitude'),
    formSire: get(state, `form.venue.sire`),
    formSiret: get(state, 'form.venue.siret'),
    formComment: get(state, 'form.venue.comment'),
    name: get(state, `form.venue.name`),
    offerer: offererSelector(state, offererId),
    user: state.user,
    venuePatch: selectVenuePatchByVenueIdByOffererId(state, venueId, offererId),
  }
}

export default compose(
  withLogin({ failRedirect: '/connexion' }),
  withRouter,
  connect(mapStateToProps)
)(RawVenuePage)
