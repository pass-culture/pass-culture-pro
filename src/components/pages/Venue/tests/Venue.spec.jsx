import React from 'react'
import Venue from '../Venue'
import { shallow } from 'enzyme'
import HeroSection from 'components/layout/HeroSection'
import { CancelButton, Field, Form, SubmitButton } from 'pass-culture-shared'
import { NavLink } from 'react-router-dom'
import VenueProvidersManager from '../VenueProvidersManager'

describe('src | components | pages | Venue | Venue', () => {
  let dispatch
  let push
  let props

  beforeEach(() => {
    dispatch = jest.fn()
    push = jest.fn()
    props = {
      currentUser: {},
      dispatch: dispatch,
      formComment: null,
      formLatitude: 5.15981,
      formLongitude: -52.63452,
      formSiret: '22222222411111',
      history: {
        location: {
          pathname: '/fake',
        },
        push: push,
      },
      location: {
        search: '',
      },
      match: {
        params: {
          offererId: 'APEQ',
          venueId: 'AQYQ',
        },
      },
      name: 'Maison de la Brique',
      offerer: {
        id: 'BQ',
        name: 'Maison du chocolat',
      },
      user: {},
      venuePatch: {
        id: null,
      },
    }
  })

  describe('snapshot', () => {
    it('should match snapshot', () => {
      // when
      const wrapper = shallow(<Venue {...props} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('render', () => {
    it('should render component with default state', () => {
      // when
      const wrapper = shallow(<Venue {...props} />)

      // then
      expect(wrapper.state('isLoading')).toBe(false)
      expect(wrapper.state('isNew')).toBe(false)
      expect(wrapper.state('isRead')).toBe(true)
    })

    it('should not render a Form when venue is virtual', () => {
      // given
      props.venuePatch = {
        isVirtual: true,
      }

      // when
      const wrapper = shallow(<Venue {...props} />)

      // then
      const form = wrapper.find(Form)
      expect(form).toHaveLength(0)
    })

    describe('when creating', () => {
      beforeEach(() => {
        props.match = {
          params: {
            offererId: 'APEQ',
            venueId: 'nouveau',
          },
        }
      })

      it('should render component with correct state values', () => {
        // when
        const wrapper = shallow(<Venue {...props} />)

        // then
        expect(wrapper.state('isEdit')).toBe(false)
        expect(wrapper.state('isLoading')).toBe(false)
        expect(wrapper.state('isNew')).toBe(true)
        expect(wrapper.state('isRead')).toBe(false)
      })

      it('should display a paragraph with proper title', () => {
        // when
        const wrapper = shallow(<Venue {...props} />)

        // then
        const heroSection = wrapper.find(HeroSection)
        expect(heroSection.find('p')).toBeDefined()
        expect(heroSection.find('p').prop('className')).toBe('subtitle')
        expect(heroSection.find('p').text()).toBe(
          'Ajoutez un lieu où accéder à vos offres.'
        )
      })

      it('should build the proper backTo link', () => {
        // when
        const wrapper = shallow(<Venue {...props} />)

        // then
        expect(wrapper.prop('backTo')).toEqual({
          label: 'Maison du chocolat',
          path: '/structures/APEQ',
        })
      })

      it('should display a hidden Field component', () => {
        // when
        const wrapper = shallow(<Venue {...props} />)

        // then
        const field = wrapper
          .find(Form)
          .find(Field)
          .first()
        expect(field).toBeDefined()
        expect(field.prop('type')).toBe('hidden')
        expect(field.prop('name')).toBe('managingOffererId')
      })

      it('should display a CancelButton component', () => {
        // when
        const wrapper = shallow(<Venue {...props} />)

        // then
        const cancelButton = wrapper.find(CancelButton)
        expect(cancelButton).toBeDefined()
        expect(cancelButton.prop('to')).toBe('/structures/APEQ')
      })

      it('should display a SubmitButton component', () => {
        // when
        const wrapper = shallow(<Venue {...props} />)

        // then
        const submitButton = wrapper.find(SubmitButton)
        expect(submitButton.dive().text()).toBe('Créer')
      })

      it('should display a comment Field component', () => {
        // when
        const wrapper = shallow(<Venue {...props} />)

        // then
        const firstFieldGroup = wrapper.find('.field-group').first()
        const lastField = firstFieldGroup.find(Field).last()
        expect(lastField).toBeDefined()
        expect(lastField.prop('label')).toBe('Commentaire (si pas de SIRET)')
        expect(lastField.prop('name')).toBe('comment')
        expect(lastField.prop('type')).toBe('textarea')
        expect(lastField.prop('required')).toBe(false)
      })

      it('should not display a NavLink component', () => {
        // when
        const wrapper = shallow(<Venue {...props} />)

        // then
        const heroSection = wrapper.find(HeroSection)
        expect(heroSection.find(NavLink)).toHaveLength(0)
      })

      it('should not display a VenueProvidersManager component', () => {
        // when
        const wrapper = shallow(<Venue {...props} />)

        // then
        expect(wrapper.find(VenueProvidersManager)).toHaveLength(0)
      })
    })

    describe('when editing', () => {
      beforeEach(() => {
        props.location = {
          search: '?modifie',
        }
        props.match = {
          params: {
            offererId: 'APEQ',
            venueId: 'AQYQ',
          },
        }
      })

      it('should render component with correct state values', () => {
        // when
        const wrapper = shallow(<Venue {...props} />)

        // then
        expect(wrapper.state('isEdit')).toBe(true)
        expect(wrapper.state('isLoading')).toBe(false)
        expect(wrapper.state('isNew')).toBe(false)
        expect(wrapper.state('isRead')).toBe(false)
      })
    })

    describe('when reading', () => {
      it('should render component with correct state values', () => {
        // when
        const wrapper = shallow(<Venue {...props} />)

        // then
        expect(wrapper.state('isEdit')).toBe(false)
        expect(wrapper.state('isLoading')).toBe(false)
        expect(wrapper.state('isNew')).toBe(false)
        expect(wrapper.state('isRead')).toBe(true)
      })
    })
  })

  describe('when requesting API', () => {
    describe('handleSuccess', () => {
      describe('when creating a venue', () => {
        const action = {
          config: {
            apiPath: '/venues/CM',
            method: 'POST',
          },
          payload: {
            datum: {
              id: 'CM',
            },
          },
        }

        it('should redirect to venue details', () => {
          // given
          const wrapper = shallow(<Venue {...props} />)
          const state = wrapper.state()

          // when
          wrapper.instance().handleSuccess(state, action)

          // then
          expect(push).toHaveBeenCalledWith('/structures/BQ/lieux/CM')
        })

        it('should dispatch a success message with valid message when venue is created', () => {
          // given
          const wrapper = shallow(<Venue {...props} />)
          const state = wrapper.state()

          // when
          wrapper.instance().handleSuccess(state, action)

          // then
          expect(dispatch).toHaveBeenCalled()
        })
      })

      describe('when editing a venue', () => {
        const action = {
          config: {
            apiPath: '/venues/CM',
            method: 'PATCH',
          },
          payload: {
            datum: {
              id: 'CM',
            },
          },
        }

        it('should redirect to venue details', () => {
          // given
          const wrapper = shallow(<Venue {...props} />)
          const state = wrapper.state()

          // when
          wrapper.instance().handleSuccess(state, action)

          // then
          expect(push).toHaveBeenCalledWith('/structures/BQ/lieux/CM')
        })

        it('should dispatch a success message with valid message when venue is modified', () => {
          // given
          const wrapper = shallow(<Venue {...props} />)
          const state = wrapper.state()

          // when
          wrapper.instance().handleSuccess(state, action)

          // then
          expect(dispatch).toHaveBeenCalledWith({
            patch: {
              text: 'Lieu modifié avec succès !',
              type: 'success',
            },
            type: 'SHOW_NOTIFICATION',
          })
        })
      })
    })

    describe('handleDataRequest', () => {
      it('should dispatch actions to update existing venue', () => {
        // given
        const wrapper = shallow(<Venue {...props} />)

        // when
        wrapper.instance().handleDataRequest(jest.fn(), jest.fn())

        // then
        expect(dispatch.mock.calls[0][0]).toEqual({
          config: {
            apiPath: '/offerers/APEQ',
            handleFail: expect.any(Function),
            handleSuccess: expect.any(Function),
            method: 'GET',
            normalizer: {
              managedVenues: {
                normalizer: { offers: 'offers' },
                stateKey: 'venues',
              },
            },
          },
          type: 'REQUEST_DATA_GET_/OFFERERS/APEQ',
        })
        expect(dispatch.mock.calls[1][0]).toEqual({
          config: { apiPath: '/userOfferers/APEQ', method: 'GET' },
          type: 'REQUEST_DATA_GET_/USEROFFERERS/APEQ',
        })
      })
    })
  })
})
