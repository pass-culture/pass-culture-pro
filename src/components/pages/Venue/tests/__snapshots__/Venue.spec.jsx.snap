// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Venue snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Venue
    currentUser={Object {}}
    dispatch={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "config": Object {
                "apiPath": "/offerers/APEQ",
                "handleSuccess": [Function],
                "method": "GET",
                "normalizer": Object {
                  "managedVenues": Object {
                    "normalizer": Object {
                      "offers": "offers",
                    },
                    "stateKey": "venues",
                  },
                },
              },
              "type": "REQUEST_DATA_GET_/OFFERERS/APEQ",
            },
          ],
          Array [
            Object {
              "config": Object {
                "apiPath": "/userOfferers/APEQ",
                "method": "GET",
              },
              "type": "REQUEST_DATA_GET_/USEROFFERERS/APEQ",
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    formInitialValues={
      Object {
        "id": "CM",
      }
    }
    handleInitialRequest={[Function]}
    handleSubmitRequest={[Function]}
    handleSubmitRequestFail={[Function]}
    handleSubmitRequestSuccess={[Function]}
    history={
      Object {
        "location": Object {
          "pathname": "/fake",
        },
        "push": [MockFunction],
      }
    }
    location={
      Object {
        "search": "",
      }
    }
    match={
      Object {
        "params": Object {
          "offererId": "APEQ",
          "venueId": "AQYQ",
        },
      }
    }
    offerer={
      Object {
        "id": "BQ",
        "name": "Maison du chocolat",
      }
    }
    query={
      Object {
        "changeToReadOnly": [MockFunction],
        "context": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "backTo": Object {
        "label": "Maison du chocolat",
        "path": "/structures/APEQ",
      },
      "children": Array [
        <HeroSection
          title="Lieu"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="cta button is-primary"
            to="/offres/creation?lieu=CM"
          >
            <span
              className="icon"
            >
              <Icon
                svg="ico-offres-w"
              />
            </span>
            <span>
              Créer une offre
            </span>
          </NavLink>
        </HeroSection>,
        <withRouter(Connect(VenueProvidersManager))
          venue={
            Object {
              "id": "CM",
            }
          }
        />,
        <ReactFinalForm
          decorators={
            Array [
              [Function],
              [Function],
              [Function],
            ]
          }
          initialValues={
            Object {
              "id": "CM",
            }
          }
          name="venue"
          onSubmit={[Function]}
          render={[Function]}
        />,
      ],
      "handleDataRequest": [Function],
      "name": "venue",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            undefined,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="cta button is-primary"
              to="/offres/creation?lieu=CM"
            >
              <span
                className="icon"
              >
                <Icon
                  svg="ico-offres-w"
                />
              </span>
              <span>
                Créer une offre
              </span>
            </NavLink>,
          ],
          "subtitle": undefined,
          "title": "Lieu",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": Array [
                <span
                  className="icon"
                >
                  <Icon
                    svg="ico-offres-w"
                  />
                </span>,
                <span>
                  Créer une offre
                </span>,
              ],
              "className": "cta button is-primary",
              "to": "/offres/creation?lieu=CM",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Icon
                    svg="ico-offres-w"
                  />,
                  "className": "icon",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "svg": "ico-offres-w",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Créer une offre",
                },
                "ref": null,
                "rendered": "Créer une offre",
                "type": "span",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "venue": Object {
            "id": "CM",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "decorators": Array [
            [Function],
            [Function],
            [Function],
          ],
          "initialValues": Object {
            "id": "CM",
          },
          "name": "venue",
          "onSubmit": [Function],
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "backTo": Object {
          "label": "Maison du chocolat",
          "path": "/structures/APEQ",
        },
        "children": Array [
          <HeroSection
            title="Lieu"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="cta button is-primary"
              to="/offres/creation?lieu=CM"
            >
              <span
                className="icon"
              >
                <Icon
                  svg="ico-offres-w"
                />
              </span>
              <span>
                Créer une offre
              </span>
            </NavLink>
          </HeroSection>,
          <withRouter(Connect(VenueProvidersManager))
            venue={
              Object {
                "id": "CM",
              }
            }
          />,
          <ReactFinalForm
            decorators={
              Array [
                [Function],
                [Function],
                [Function],
              ]
            }
            initialValues={
              Object {
                "id": "CM",
              }
            }
            name="venue"
            onSubmit={[Function]}
            render={[Function]}
          />,
        ],
        "handleDataRequest": [Function],
        "name": "venue",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              undefined,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="cta button is-primary"
                to="/offres/creation?lieu=CM"
              >
                <span
                  className="icon"
                >
                  <Icon
                    svg="ico-offres-w"
                  />
                </span>
                <span>
                  Créer une offre
                </span>
              </NavLink>,
            ],
            "subtitle": undefined,
            "title": "Lieu",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": Array [
                  <span
                    className="icon"
                  >
                    <Icon
                      svg="ico-offres-w"
                    />
                  </span>,
                  <span>
                    Créer une offre
                  </span>,
                ],
                "className": "cta button is-primary",
                "to": "/offres/creation?lieu=CM",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Icon
                      svg="ico-offres-w"
                    />,
                    "className": "icon",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "svg": "ico-offres-w",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Créer une offre",
                  },
                  "ref": null,
                  "rendered": "Créer une offre",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "venue": Object {
              "id": "CM",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "decorators": Array [
              [Function],
              [Function],
              [Function],
            ],
            "initialValues": Object {
              "id": "CM",
            },
            "name": "venue",
            "onSubmit": [Function],
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
